sidecar/internal/reconciler/driver.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/util/validation"
+
+	cosiproto "sigs.k8s.io/container-object-storage-interface/proto"
+)
+
+// DriverInfo contains critical info about the paired driver that is needed by all reconcilers
+type DriverInfo struct {
+	name               string
+	supportedProtocols []cosiproto.ObjectProtocol_Type
+
+	provisionerClient cosiproto.ProvisionerClient
+}
```

Comment:
This DriverInfo will later be passed to all sidecar reconcilers so that they can query the provisioner driver server. For now, it's constructor is used to validate the driver info response.

Also, CSI internally uses the proto data structures to store info about the system, so it seemed appropriate to follow that pattern here. Alternatives would be to use the Kube API types or a separate internal type.